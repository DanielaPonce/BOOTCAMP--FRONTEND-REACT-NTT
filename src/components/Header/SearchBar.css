@media only screen and (max-width: 739px) {

	.header-container header .search-bar,
	.header-container header .search-bar input,
	.search-container select,
	.search-container {
		width: 100%;
		margin: 0;
	}

	.search-container {
		flex-direction: column;
	}
}
