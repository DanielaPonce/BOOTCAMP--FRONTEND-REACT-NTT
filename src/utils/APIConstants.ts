export enum APIConstants {
	API_BASE_URL = 'https://dummyjson.com'
}
