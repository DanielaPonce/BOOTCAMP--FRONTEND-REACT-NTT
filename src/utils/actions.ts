export enum Actions {
	ADD_TO_CART_TYPE = 'ADD_TO_CART',
	REMOVE_FROM_CART_TYPE = 'REMOVE_FROM_CART',
	UPDATE_QUANTITY_TYPE = 'UPDATE_QUANTITY',
	CLEAR_CART_TYPE = 'CLEAR_CART'
}
