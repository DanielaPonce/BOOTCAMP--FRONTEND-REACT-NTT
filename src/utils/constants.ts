export enum Constants {
	API_BASE_URL = 'https://dummyjson.com',
	appTitle = 'My Market',
	searchBarPlaceholder = 'Buscar productos...',
	defaultSelectValue = 'all',
	ADD_TO_CART_TYPE = 'ADD_TO_CART'
}
