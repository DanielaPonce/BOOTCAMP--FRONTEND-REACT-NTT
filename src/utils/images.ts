export enum Images {
	Logo = '/src/assets/logo/logo.webp'
}
