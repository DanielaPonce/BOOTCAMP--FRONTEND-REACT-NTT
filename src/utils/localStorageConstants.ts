export enum LocalStorageConstants {
	CartKey = 'cart'
}
