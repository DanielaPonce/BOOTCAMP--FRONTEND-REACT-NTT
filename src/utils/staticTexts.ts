export enum StaticTexts {
	appTitle = 'My Market',
	searchBarPlaceholder = 'Buscar productos...',
	defaultSelectValue = 'all'
}
