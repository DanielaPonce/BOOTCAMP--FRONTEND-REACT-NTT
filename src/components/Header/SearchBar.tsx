import { FC } from 'react';
import { Constants } from '../../utils/constants';
import { StaticTexts } from '../../utils/staticTexts';

type Props = {
	onInputSearch: (search: string) => void;
};

const SearchBar: FC<Props> = ({ onInputSearch }) => {
	return (
		<div className="search-bar">
			<input
				type="search"
				placeholder={StaticTexts.searchBarPlaceholder}
				onInput={(e) =>
					onInputSearch((e.target as HTMLInputElement).value)
				}
			/>
		</div>
	);
};

export default SearchBar;
