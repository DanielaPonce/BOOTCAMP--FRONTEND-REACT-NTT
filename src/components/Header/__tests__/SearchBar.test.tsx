import { jest } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import { StaticTexts } from '../../../utils/staticTexts';
import SearchBar from '../SearchBar';

describe('SearchBar Component', () => {
	const mockOnInputSearch = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders correctly', () => {
		render(<SearchBar onInputSearch={mockOnInputSearch} />);

		const input = screen.getByRole('searchbox');
		expect(input).toBeInTheDocument();
		expect(input).toHaveAttribute('type', 'search');
		expect(input).toHaveAttribute(
			'placeholder',
			StaticTexts.SearchBarPlaceholder
		);
	});

	test('calls onInputSearch when typing in the search input', () => {
		render(<SearchBar onInputSearch={mockOnInputSearch} />);

		const input = screen.getByRole('searchbox');
		fireEvent.input(input, { target: { value: 'test query' } });

		expect(mockOnInputSearch).toHaveBeenCalledTimes(1);
		expect(mockOnInputSearch).toHaveBeenCalledWith('test query');
	});

	test('matches snapshot', () => {
		const { asFragment } = render(
			<SearchBar onInputSearch={mockOnInputSearch} />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
