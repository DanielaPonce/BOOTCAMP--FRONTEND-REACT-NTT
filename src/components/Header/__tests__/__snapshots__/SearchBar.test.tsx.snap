// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-bar"
  >
    <input
      placeholder="Buscar productos..."
      type="search"
    />
  </div>
</DocumentFragment>
`;
